src/devices/Mpu9250/Mpu9250.cs

```diff
@@ -0,0 +1,983 @@
+                        val = 250.0f;
+                        break;
+                    case GyroscopeRange.Range0500Dps:
+                        val = 500.0f;
+                        break;
+                    case GyroscopeRange.Range1000Dps:
+                        val = 1000.0f;
+                        break;
+                    case GyroscopeRange.Range2000Dps:
+                        val = 2000.0f;
+                        break;
+                    default:
+                        break;
+                }
+
+                val /= Adc;
+                // the sample rate diver only apply for the non FS modes
+                if ((GyroscopeBandwidth != GyroscopeBandwidth.Bandwidth3600HzFS32) &&
+                    (GyroscopeBandwidth != GyroscopeBandwidth.Bandwidth8800HzFS32))
+                    return val / (1 + SampleRateDivider);
+                return val;
+            }
+        }
+
+        /// <summary>
+        /// Get the gyroscope in degrees per seconds
+        ///    +Z   +Y
+        ///  \  |  /
+        ///   \ | /
+        ///    \|/
+        ///    /|\
+        ///   / | \
+        ///  /  |  \
+        ///         +X
+        /// </summary>
+        public Vector3 Gyroscope => GetRawGyroscope() * GyroscopeConvertion;
+
+        private Vector3 GetRawGyroscope()
+        {
+            Span<byte> rawData = stackalloc byte[6] { 0, 0, 0, 0, 0, 0 };
+            Vector3 gyro = new Vector3();
+            ReadByteArray(Register.GYRO_XOUT_H, rawData);
+            gyro.X = BinaryPrimitives.ReadInt16BigEndian(rawData);
+            gyro.Y = BinaryPrimitives.ReadInt16BigEndian(rawData.Slice(2));
+            gyro.Z = BinaryPrimitives.ReadInt16BigEndian(rawData.Slice(4));
+            return gyro;
+        }
+
+        #endregion
+
+        #region Temperature
+
+        /// <summary>
+        /// Get the temperature
+        /// </summary>
+        public Temperature Temperature
+        {
+            get
+            {
+                Span<byte> rawData = stackalloc byte[2] { 0, 0 };
+                ReadByteArray(Register.TEMP_OUT_H, rawData);
+                // formula from the documentation
+                return Temperature.FromCelsius((BinaryPrimitives.ReadInt16BigEndian(rawData) - 21) / 333.87 + 21);
+            }
+        }
+
+        #endregion
+
+        #region Modes, constructor, Dispose
+
+        /// <summary>
+        /// Setup the Wake On Motion. This mode generate a rising signal on pin INT
+        /// You can catch it with a normal GPIO and place an interruption on it if supprted
+        /// Reading the sensor won't give any value until it wakes up preriodically
+        /// Only Accelerator data is availablein this mode
+        /// </summary>
+        /// <param name="magnetometerThreshold"></param>
+        /// <param name="acceleratorLowPower"></param>
+        public void SetWakeOnMotion(uint magnetometerThreshold, AccelerometerLowPowerFrequency acceleratorLowPower)
```

Comment:
Do you actually need the threshold to be uint? if not I would just change it to int

Reply:
It makes sense to use `uint` if e.g. for physical reasons the parameter can only accept positive values. I therefore consider the choice not critical.